src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -311,13 +311,36 @@ public StringIndirectionNode StringIndirection(string data)
             return _stringIndirectionNodes.GetOrAdd(data);
         }
 
-        public ArrayOfEmbeddedDataNode GCStaticsRegion = new ArrayOfEmbeddedDataNode("__GCStaticRegionStart", "__GCStaticRegionEnd", null);
-        public ArrayOfEmbeddedDataNode ThreadStaticsRegion = new ArrayOfEmbeddedDataNode("__ThreadStaticRegionStart", "__ThreadStaticRegionEnd", null);
-        public ArrayOfEmbeddedDataNode StringTable = new ArrayOfEmbeddedDataNode("__str_fixup", "__str_fixup_end", null);
+        /// <summary>
+        /// Returns alternative symbol name that object writer should produce for given symbols
+        /// in addition to the regular one.
+        /// </summary>
+        public string GetSymbolAltenateName(ISymbolNode node)
+        {
+            string value;
+            if (!NodeAliases.TryGetValue(node, out value))
+                return null;
+            return value;
+        }
+
+        public ArrayOfEmbeddedDataNode GCStaticsRegion = new ArrayOfEmbeddedDataNode(
+            NameMangler.CompilationUnitPrefix + "__GCStaticRegionStart", 
+            NameMangler.CompilationUnitPrefix + "__GCStaticRegionEnd", 
+            null);
+        public ArrayOfEmbeddedDataNode ThreadStaticsRegion = new ArrayOfEmbeddedDataNode(
+            NameMangler.CompilationUnitPrefix + "__ThreadStaticRegionStart",
+            NameMangler.CompilationUnitPrefix + "__ThreadStaticRegionEnd", 
+            null);
+        public ArrayOfEmbeddedDataNode StringTable = new ArrayOfEmbeddedDataNode(
+            NameMangler.CompilationUnitPrefix + "__str_fixup",
+            NameMangler.CompilationUnitPrefix + "__str_fixup_end", 
+            null);
 
         public Dictionary<TypeDesc, List<MethodDesc>> VirtualSlots = new Dictionary<TypeDesc, List<MethodDesc>>();
 
-        public static NameMangler NameMangler;
+        public Dictionary<ISymbolNode, string> NodeAliases = new Dictionary<ISymbolNode, string>();
```

Comment:
When I wanted to clean up issue #153, I had this in mind:
Alias(es) can be a member of ISymbolNode, node.GetAliases() and emit symbol def for each of those.

Please resolve Issue #153.